src/summary.rs

```diff
@@ -65,18 +66,24 @@ impl SummaryFile {
     /// if user_space_pledged is provided, it creates a new summary file
     /// else, it tries to open the existing summary file
     #[instrument]
-    pub(crate) async fn new(user_space_pledged: Option<ByteSize>) -> Result<SummaryFile> {
+    pub(crate) async fn new(maybe_user_space_pledged: Option<ByteSize>) -> Result<SummaryFile> {
         let summary_path = summary_path();
         let summary_dir = summary_dir();
 
-        let mut summary_file;
-        // providing `Some` value for `user_space_pledged` means, we are creating a new
-        // file, so, first check if the file exists to not erase its content
-        if let Some(user_space_pledged) = user_space_pledged {
-            // File::create will truncate the existing file, so first
-            // check if the file exists, if not, `open` will return an error
-            // in this case, create the file and necessary directories
-            if File::open(&summary_path).await.is_err() {
+        let file_handle = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .open(&summary_path)
+            .await
+            .context("couldn't open existing summary file");
+        match file_handle {
+            Err(e) => {
```

Comment:
Why would you still try to write into the file in case of error during opening? I do not understand.

Old code makes not that much sense either. What needs to happen instead is you need to write to a new file and rename it back into intended name. This way you will never get truncated file in the first place.